src/modules/model.js

```diff
@@ -66,6 +74,21 @@ class Model {
     });
   }
 
+  // eslint-disable-next-line no-underscore-dangle
+  _generateId(data, providedId) {
+    if ((providedId === null) === (this.primaryKeys === null)) {
+      throw new MustProvideIdXorPrimaryKeys();
+    }
+    if (this.primaryKeys !== null && this.primaryKeys.some(k => data[k] === undefined)) {
+      throw new IncompletePrimaryKey();
+    }
+    return providedId !== null
+      ? providedId
+      : objectHash(this.primaryKeys
+        .sort()
```

Comment:
I don't think objectHash differentiates object key order?

Reply:
Ah, you're right. They sort by default, I only saw the option